common/djangoapps/student/models.py

```diff
@@ -1602,6 +1602,11 @@ def enforce_single_login(sender, request, user, signal, **kwargs):    # pylint:
         else:
             key = None
         if user:
+            if not UserProfile.objects.filter(user=user):
```

Comment:
I don't think users on edx can change their usernames.

I'm willing to let it slide, but at minimum, that `filter` should be: `if not UserProfile.objects.filter(user=user).exists():`